crates/egui-winit/src/window_settings.rs

```diff
@@ -86,4 +83,58 @@ impl WindowSettings {
             *size = size.at_most(max_size);
         }
     }
+
+    pub fn clamp_window_to_sane_position<E>(
+        &mut self,
+        event_loop: &winit::event_loop::EventLoopWindowTarget<E>,
+    ) {
+        use std::ops::{Add, AddAssign, MulAssign, Sub};
+
+        if let (Some(position), Some(inner_size)) = (&mut self.position, &self.inner_size_points) {
+            let monitors = event_loop.available_monitors();
+            // default to primary monitor, in case the correct monitor was disconnected.
+            let mut monitor: winit::monitor::MonitorHandle = match event_loop.primary_monitor() {
+                Some(m) => m,
+                // No primary monitor, grab index 0 instead
+                _ => {
+                    match event_loop.available_monitors().nth(0) {
+                        Some(m) => m,
+                        None => return, // No monitors?
+                    }
+                }
+            };
+            for _monitor in monitors {
```

Comment:
The `_` prefix shouldn't be used to avoid name clashes.

Maybe call the outer `monitor` `active_monitor` or something?

Reply:
Sorry about that, looks like I missed renaming that one from prototyping.